client/src/cli/util.rs

```diff
@@ -0,0 +1,45 @@
+#[macro_export]
+macro_rules! register_command {
+    ( $({$name:ident, $handler:ident}), *) => {
+        use clap::{Parser, Subcommand};
+
+        /// The overall command line struct
+        #[derive(std::fmt::Debug, Parser)]
+        #[command(
+            name = "ipc",
+            about = "The IPC node command line tool",
+            version = "v0.0.1"
+        )]
+        #[command(propagate_version = true)]
+        struct IPCNode {
+            #[command(subcommand)]
+            command: Commands,
+        }
+
+        /// The subcommand to be called
+        #[derive(Debug, Subcommand)]
+        enum Commands {
+            $(
+                $name(<$handler as CommandLineHandler>::Request),
+            )*
+
+        }
+
+        pub async fn cli() {
+            let args = IPCNode::parse();
+            let r = match &args.command {
```

Comment:
Isn't this possible with an enum instead of a macro?

Reply:
wouldn't a macro be easier to use? The declaration and routing is done behind the scene so that only one place is needed to be updated.